packages/amplify-graphql-transformer-core/src/types/utils.ts

```diff
@@ -0,0 +1,229 @@
+  } else if (resource.cfnResourceType === 'AWS::AppSync::FunctionConfiguration') {
+    return resource as CfnFunctionConfiguration;
+  } else if (resource.cfnResourceType === 'AWS::Lambda::Function') {
+    return resource as CfnFunction;
+  } else if (resource.cfnResourceType === 'AWS::Lambda::EventSourceMapping') {
+    return resource as CfnEventSourceMapping;
+  } else if (resource.cfnResourceType === 'AWS::Elasticsearch::Domain') {
+    return resource;
+  } else {
+    return undefined;
+  }
+};
+
+export const convertToAppsyncResourceObj = (amplifyObj: any) => {
+  let appsyncResourceObject: AppSyncServiceResourceStack = {};
+  Object.keys(amplifyObj).forEach(keys => {
+    if (keys === 'api') {
+      appsyncResourceObject.api = amplifyObj.api;
+    } else if (keys === 'models' && !_.isEmpty(amplifyObj[keys])) {
+      // require filter using keyName
+      appsyncResourceObject.models = {};
+      Object.keys(amplifyObj.models).forEach(key => {
+        appsyncResourceObject.models![key] = generateModelDirectiveObject(amplifyObj.models[key]);
+      });
+    } else if (keys === 'function' && !_.isEmpty(amplifyObj[keys])) {
+      appsyncResourceObject.function = {};
+      const functionStackObj = amplifyObj.function.FunctionDirectiveStack;
+      appsyncResourceObject.function = generateFunctionDirectiveObject(functionStackObj);
+    } else if (keys === 'http' && !_.isEmpty(amplifyObj[keys])) {
+      appsyncResourceObject.http = {};
+      const httpStackObj = amplifyObj.http.HttpStack;
+      appsyncResourceObject.http = generateHttpDirectiveObject(httpStackObj);
+    } else if (keys === 'openSearch' && !_.isEmpty(amplifyObj[keys])) {
+      appsyncResourceObject.opensearch = {};
+      const openSearchStackObj = amplifyObj.openSearch.SearchableStack;
+      appsyncResourceObject.opensearch = generateOpenSearchDirectiveObject(openSearchStackObj);
+    } else if (keys === 'predictions' && !_.isEmpty(amplifyObj[keys])) {
+      appsyncResourceObject.predictions = {};
+      appsyncResourceObject.predictions = amplifyObj.predictions.PredictionsDirectiveStack;
+      if (!_.isEmpty(amplifyObj.predictions.PredictionsDirectiveStack['predictionsLambda.handler'])) {
+        appsyncResourceObject.predictions!.predictionsLambdaFunction =
+          amplifyObj.predictions.PredictionsDirectiveStack['predictionsLambda.handler'];
+      }
+    }
+  });
+  return appsyncResourceObject;
+};
+
+const generateFunctionDirectiveObject = (functionStackObj: any) => {
+  let functionObj: Partial<FunctionDirectiveStack & AppsyncStackCommon> = {};
+  Object.keys(functionStackObj).forEach(key => {
+    if (key.includes('resolver')) {
+      functionObj.resolvers = functionStackObj.resolvers;
+    } else if (key.includes('appsyncFunctions')) {
+      functionObj.appsyncFunctions = functionStackObj.appsyncFunctions;
+    } else if (functionStackObj[key].cfnResourceType.includes('DataSource')) {
+      if (!functionObj.lambdaDataSource) {
```

Comment:
use _.set() here as well (any anywhere else potentially undefined intermediate paths need to be set in an object)

Reply:
I will attend this when merging to master . Need to get the tagged release off